src/com/hms_networks/americas/sc/realtimedata/RealTimeDataQueueManager.java

```diff
@@ -148,9 +149,20 @@ public static void dataFetcher(int tagGroup) {
     ArrayList tagGroupArray = (ArrayList) tagManagers.get(tagGroup);
     if (tagGroupArray != null) {
       // for each tag in the tag group, record a new value
-      for (int i = 0; i < tagGroupArray.size(); i++) {
-        TagInfo currentTag = ((TagInfo) ((ArrayList) tagGroupList.get(tagGroup)).get(i));
-        ((RealTimeTagDataPointManager) tagGroupArray.get(i)).recordCurentTagValue(currentTag);
+      for (int iterator = 0; iterator < tagGroupArray.size(); iterator++) {

```

Comment:
wouldnt changing iterator to something like tagNum make the code a lot more readable.